account_sdk/src/deploy_contract/deployment.rs

```diff
@@ -0,0 +1,69 @@
+use starknet::{
+    accounts::SingleOwnerAccount,
+    contract::ContractFactory,
+    core::types::{FieldElement, InvokeTransactionResult},
+    providers::{JsonRpcClient, Provider},
+    signers::Signer,
+};
+
+use crate::suppliers::{PredeployedClientSupplier, RpcClientSupplier};
+
+use super::pending::PendingDeployment;
+
+pub struct CustomAccountDeployment<'a, Q> {
+    supplier: &'a Q,
+}
+
+impl<'a, Q> CustomAccountDeployment<'a, Q> {
+    pub fn new<T>(client_supplier: &'a Q) -> Self
+    where
+        Q: RpcClientSupplier<T> + PredeployedClientSupplier,
+        JsonRpcClient<T>: Provider,
+        T: Send + Sync,
+    {
+        CustomAccountDeployment {
+            supplier: &client_supplier,
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct DeployResult {
+    pub deployed_address: FieldElement,
+    pub transaction_hash: FieldElement,
+}
+
+impl<'a, Q> CustomAccountDeployment<'a, Q> {
```

Comment:
```suggestion
impl<'a, Q> AccountDeployment<'a, Q> {
```

Remove all the `Custom` stuff